File: spec/cpp_stl_98/test_params_def.cpp
#include <boost/test/unit_test.hpp>

#include <params_def.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_params_def) {
    std::ifstream ifs("src/term_strz.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    params_def_t* r = new params_def_t(5, true, &ks);

    BOOST_CHECK_EQUAL(r->buf(), "foo|b");
    BOOST_CHECK_EQUAL(r->trailer(), 0x61);

    delete r;
}

File: spec/rust/tests/test_params_def.rs
use std::fs;
extern crate kaitai;
use self::kaitai::*;
use rust::formats::params_def::*;

#[test]
fn test_params_def() -> KResult<()> {
    let bytes = fs::read("../../src/term_strz.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let f = |t: &mut ParamsDef| Ok(t.set_params(5, true));
    let r: OptRc<ParamsDef> = ParamsDef::read_into_with_init(&_io, None, None, &f)?;

    assert_eq!(*r.buf(), "foo|b");
    assert_eq!(*r.trailer(), 0x61);
    Ok(())
}
